clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaRecordHelper.java

```diff
@@ -199,4 +201,18 @@ public static boolean isTtlElapsed(final List<KafkaHeader> headers) {
         }
     }
 
+    /**
+     * Gets the String value of a specific <em>header</em>.
+     * <p>
+     * If the headers contain multiple occurrences of the same key, the value of its first
+     * occurrence is returned.
+     *
+     * @param headers The headers to retrieve the value from.
+     * @param key The header key.
+     * @return The value or {@code null} if the headers do not contain a value of the expected type for the given key.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public static String getStringHeaderValue(final List<KafkaHeader> headers, final String key) {
```

Comment:
Why does this not return an optional like the other methods here?

Reply:
This method has been moved from the `RecordUtil` class (as that class has been deleted). The classes which use this utility method expect that this method returns a `String` value if present else _null_.

Reply:
I know that. But should not `KafkaRecordHelper` define its interface and code that uses it be adapted instead?

Reply:
FMPOV it is more of a taste if to return a _String_ or an _Optional_ for a utility method. Considering that we already return _Optional_ for an already existing method in that class, I find it reasonable to have the same style. If `getStringHeaderValue(...)` is to return an _Optional_, I find it there is not much value in having that method and rather better to use the `getHeaderValue(headers, key, String.class)`.  I have made changes to use the  `getHeaderValue(headers, key, String.class)`.